yodeploy/virtualenv.py

```diff
@@ -91,58 +81,39 @@ def create_ve(
         os.chdir(cwd)
 
 
-def install_requirements(ve_dir, pypi, requirements):
-    sub_log = logging.getLogger(__name__ + '.easy_install')
+def install_requirements(ve_dir, pypi, req_file):
+    sub_log = logging.getLogger(__name__ + '.pip')
 
-    if pypi is None:
-        pypi = []
-    else:
-        pypi = ['--index-url', pypi]
-
-    for requirement in requirements:
-        log.info('Preparing to install %s', requirement)
-
-        cmd = [
-            os.path.join('bin', 'python'),
-            os.path.join('bin', 'easy_install'), '--always-unzip'
-        ] + pypi + [requirement]
-
-        p = subprocess.Popen(cmd, cwd=ve_dir, stdout=subprocess.PIPE)
-        for line in iter(p.stdout.readline, b''):
-            line = line.decode('utf-8').strip()
-            sub_log.info('%s', line)
-            if line.startswith('Removing'):
-                log.error('Requirements were incompatible: %s', line)
-                sys.exit(1)
-
-        if p.wait() != 0:
-            log.error('easy_install exited non-zero (%i)', p.returncode)
-            sys.exit(1)
-
-        log.info('Installed %s', requirement)
-
-
-def check_requirements(ve_dir, requirements):
-    """Given a ve root, and a list of requirements, ensure that they are met"""
-    script = (
-        'import sys, pkg_resources\n'
-        'for req in %r:\n'
-        '  try:\n'
-        '    pkg_resources.get_distribution(req)\n'
-        '  except pkg_resources.DistributionNotFound:\n'
-        '    pass\n'
-        '  except pkg_resources.VersionConflict as e:\n'
-        '    print(\n'
-        '      "Incompatible requirement: %%s\\n" %% e)\n'
-        '    sys.exit(1)\n'
-    ) % (requirements,)
+    log.info('Installing requirements')
+
+    cmd = [
+        os.path.join('bin', 'python'), '-m', 'pip', 'install',
+        '-r', req_file,
+    ]
 
+    if pypi:
+        cmd += ['--index-url', pypi]
+
+    p = subprocess.Popen(cmd, cwd=ve_dir, stdout=subprocess.PIPE)
+    for line in iter(p.stdout.readline, b''):
+        line = line.decode('utf-8').strip()
+        sub_log.info('%s', line)
+
+    if p.wait() != 0:
+        log.error('pip exited non-zero (%i)', p.returncode)
+        sys.exit(1)
+
+
+def check_requirements(ve_dir):
+    """Run pip check"""
     p = subprocess.Popen(
-        os.path.join(ve_dir, 'bin', 'python'),
-        stdin=subprocess.PIPE, stdout=subprocess.PIPE)
-    out, _ = p.communicate(script.encode('utf-8'))
+        (os.path.join(ve_dir, 'bin', 'python'), '-m', 'pip', 'check'),
```

Comment:
Are the shebangs not respected? Looking at my local virtualenv's bins, all the shebangs have been updated. The bins don't use `/usr/bin/env` but instead point to the virtualenv's python.

No biggie though. This way is probably better, if for whatever reason the shebangs are not updated this would be incredibly difficult to debug.